src/mono/wasm/debugger/BrowserDebugProxy/MonoProxy.cs

```diff
@@ -508,6 +508,36 @@ private async Task<Result> RuntimeGetProperties(MessageId id, DotnetObjectId obj
             return res;
         }
 
+        private async Task<bool> EvaluateCondition(SessionId sessionId, ExecutionContext context, JObject mono_frame, string condition, CancellationToken token)
+        {
+            if (!string.IsNullOrEmpty(condition) && mono_frame != null)
+            {
+                try {
+                    var resolver = new MemberReferenceResolver(this, context, sessionId, mono_frame["frame_id"].Value<int>(), logger);
+
+                    JObject retValue = await resolver.Resolve(condition, token);
+                    if (retValue == null)
+                    {
+                        retValue = await EvaluateExpression.CompileAndRunTheExpression(condition, resolver, token);
+                    }
+
+                    if (retValue != null)
```

Comment:
If `retValue == null`, then this would return true?

Also, would be nice to have test cases for this kinda cases .. where 1. both steps to get `retValue` fail; 2. first fails, but second succeeds.

Reply:
The second okay, it's already tested. In the first test that you suggested I have no idea how to do the second retValue return null...

Reply:
I think the 3 cases would be:

1. first succeeds should be `obj.Field`, or `obj.Property`
2. first fails, second succeeds should be `j < 5`, or `j > obj.Property`
3. Both fail might be an invalid expression, or one containing members that we can't find at all? - And this one will throw, like you pointed out OOB.

Reply:
OKay, they are all tested.